### PR TITLE
fix: calling `get_secrets` in setup code fails

### PR DESCRIPTION
**Pull request by bmoxb**
https://github.com/shuttle-hq/shuttle/issues/197

**TODO: Handle in `Loader::load` instead of codegen** - need to work out Tokio runtime problems.

* The set of secrets read from `Secrets.toml` are now sent as a HTTP header instead of in a separate request.
* That set of initial secrets is then set immediately before the user's setup function is called.

### Considerations

* Is passing the received set of secrets through the full deployment pipeline all the way to the service build function the best approach? Is there a simpler way to achieve the say outcome?
* Secrets from `Secrets.toml` are now sent as a HTTP header with JSON encoding - this should not result in an invalid HTTP request but may need to test some edge cases (e.g., weird unicode characters). If it is a problem, then a solution could be base64 encoding.

### CODE REVIEW

File: api/src/deployment.rs [hunk 1]
```diff
@@ -49,10 +49,15 @@ impl Deployment {
         }
     }
 
-    fn from_bytes(fqdn: &str, project: ProjectName, crate_bytes: Vec<u8>) -> Self {
+    fn from_bytes(
+        fqdn: &str,
+        project: ProjectName,
+        crate_bytes: Vec<u8>,
+        initial_secrets: InitialSecrets,
```
Comment (chesedo) [inline api/src/deployment.rs:56]: Won't the `Secrets.toml` file also be in `crate_bytes`? Could we possibly read them from there on the `api` end?
